src/pages/PrintOnlyScreen.tsx

```diff
@@ -0,0 +1,130 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+
+import { Election, MarkVoterCardFunction, VotesDict } from '../config/types'
+import { NullPrinter } from '../utils/printer'
+import isEmptyObject from '../utils/isEmptyObject'
+
+import Prose from '../components/Prose'
+import Main, { MainChild } from '../components/Main'
+import PrintedBallot from '../components/PrintedBallot'
+import Loading from '../components/Loading'
+
+interface Props {
+  ballotStyleId: string
+  election: Election
+  isLiveMode: boolean
+  isVoterCardPresent: boolean
+  markVoterCardPrinted: MarkVoterCardFunction
+  precinctId: string
+  printer: NullPrinter
+  votes: VotesDict
+}
+
+const printerTimerTimeoutSeconds = 5
+const lastVotesKey = 'lastVotes'
+
+const PrintAppScreen = ({
+  ballotStyleId,
+  election,
+  isLiveMode,
+  isVoterCardPresent,
+  markVoterCardPrinted,
+  precinctId,
+  printer,
+  votes: cardVotes,
+}: Props) => {
+  let printerTimer = useRef(0)
+  const localVotes = JSON.parse(
+    window.localStorage.getItem(lastVotesKey) || '{}'
+  )
+  const isLocalVotes = !isEmptyObject(localVotes)
+  const [isPrinted, updateIsPrinted] = useState(false)
+  const isCardVotesEmpty = isEmptyObject(cardVotes)
+
+  const printBallot = useCallback(async () => {
+    const isUsed = await markVoterCardPrinted()
+    /* istanbul ignore else */
+    if (isUsed) {
+      await printer.print()
+      printerTimer.current = window.setTimeout(() => {
+        updateIsPrinted(true) // add timeout here
+      }, printerTimerTimeoutSeconds * 1000)
+    }
+  }, [markVoterCardPrinted, printer])
+
+  useEffect(() => {
+    if (!isEmptyObject(cardVotes)) {
+      window.localStorage.setItem(lastVotesKey, JSON.stringify(cardVotes))
+      printBallot()
+    }
+  }, [cardVotes, printBallot])
+
+  useEffect(() => {
+    if (!isVoterCardPresent) {
+      updateIsPrinted(false)
+    }
+  }, [isVoterCardPresent])
+
+  useEffect(() => {
+    return () => clearTimeout(printerTimer.current)
```

Comment:
Maybe this can't be avoided in this situation, but one of the selling points of hooks & `useEffect` was that you could co-locate the "register & unregister"-type actions, or in this case `setTimeout` & `clearTimeout`. Is that not possible or not desirable here?